packages/tasks_panel/src/misc/e2e-config.ts

```diff
@@ -0,0 +1,220 @@
+            }
+            if (style !== undefined) {
+              return {
+                wsFolder,
+                project: asWsRelativePath(info.path),
+                style,
+              };
+            }
+          }
+        }
+      })
+    );
+  }
+  return Promise.all(items).then((items) => compact(items));
+}
+
+export async function addTaskDefinition(wsFolder: string, tasks: TaskDefinition[]): Promise<any> {
+  return updateTasksConfiguration(
+    wsFolder,
+    concat(workspace.getConfiguration("tasks", Uri.file(wsFolder))?.get("tasks") ?? [], tasks)
+  );
+}
+
+export async function isFileExist(uri: Uri): Promise<boolean> {
+  try {
+    return !!(await workspace.fs.stat(uri));
+  } catch (e) {
+    return false;
+  }
+}
+
+export enum LabelType {
+  uniq,
+  sequence,
+}
+
+export async function generateMtaDeployTasks(
+  wsFolder: string,
+  project: string,
+  labelType = LabelType.uniq
+): Promise<TaskDefinition[]> {
+  async function populateCfDetails(): Promise<CfDetails> {
+    try {
+      const targets = await cfGetTargets();
+      if (isEmpty(targets) || (size(targets) === 1 && targets[0].label === DEFAULT_TARGET)) {
+        throw new Error("No CF targets found");
+      }
+      const targetName = find(targets, "isCurrent")?.label;
+      if (!targetName) {
+        throw new Error("No CF current target defined");
+      }
+      /* istanbul ignore next */
+      return {
+        cfTarget: targetName,
+        cfEndpoint: (await cfGetConfigFileField("Target", targetName)) ?? "",
+        cfOrg: (await cfGetConfigFileField("OrganizationFields", targetName))?.Name ?? "",
+        cfSpace: (await cfGetConfigFileField("SpaceFields", targetName))?.Name ?? "",
+      };
+    } catch (e: any) {
+      getLogger().debug(`Can not populate cf target details`, { reason: e.toString() });
+      return { cfTarget: "", cfEndpoint: "", cfOrg: "", cfSpace: "" };
+    }
+  }
+  const projectUri = Uri.joinPath(Uri.file(wsFolder), project);
+  let label = `Build MTA`;
+  const taskBuild = {
+    type: "build.mta",
+    label: labelType === LabelType.uniq ? getUniqueTaskLabel(label) : label,
+    taskType: "Build",
+    projectPath: `${projectUri.fsPath}`,
+    extensions: [],
+  };
+  label = `Deploy MTA to Cloud Foundry`;
+  const taskDeploy = extend(
+    {
+      type: "deploy.mta.cf",
+      label: labelType === LabelType.uniq ? getUniqueTaskLabel(label) : label,
+      taskType: "Deploy",
+      mtarPath: `${projectUri.fsPath}/mta_archives/${project || last(compact(split(wsFolder, path.sep)))}_0.0.1.mtar`,
+      extensions: [],
+      dependsOn: [`${taskBuild.label}`],
+    },
+    await populateCfDetails()
+  );
+
+  return [taskBuild, taskDeploy];
+}
+
+export function isTasksSettled(wsFolder: string, _tasks: TaskDefinition[]): boolean {
+  const tasks: TaskDefinition[] = workspace.getConfiguration("tasks", Uri.file(wsFolder)).get("tasks") ?? [];
+  return reduce(
+    _tasks,
+    (acc, task) => {
```

Comment:
`_tasks` and `task` variable names are confusing

Reply:
done